framework/src/main/java/org/checkerframework/framework/util/dependenttypes/DependentTypesHelper.java

```diff
@@ -1083,9 +1116,9 @@ protected Void combineRs(Void r1, Void r2) {
     }
```

Comment:
Just above in `combineRs`, I was trying to figure out the relationship between `p` and `R`.  I think they are unrelated.

In `AnnotatedTypeComparer`, could you document `R`?  I think it's the type of the value returned by the scanner.  In that case, `combineRs` would be better renamed to `combineResults` or the like.  Please do all that in a separate pull request from this one, and it will probably land before this one.

Reply:
I've taken a note to do this.